lightning/src/routing/router.rs

```diff
@@ -373,11 +373,13 @@ pub fn get_route<L: Deref>(our_node_id: &PublicKey, network: &NetworkGraph, paye
 	if final_value_msat == 0 {
 		return Err(LightningError{err: "Cannot send a payment of 0 msat".to_owned(), action: ErrorAction::IgnoreError});
 	}
+	let last_hops = last_hops.iter().map(|hops| hops.0.clone()).collect::<Vec<_>>();
 
-	let last_hops = last_hops.iter().filter_map(|hops| hops.0.last()).collect::<Vec<_>>();
-	for last_hop in last_hops.iter() {
-		if last_hop.src_node_id == *payee {
-			return Err(LightningError{err: "Last hop cannot have a payee as a source.".to_owned(), action: ErrorAction::IgnoreError});
+	for routes in last_hops.iter() {
+		for last_hop in routes.iter() {
+			if last_hop.src_node_id == *payee {
```

Comment:
AFAIU we already support multiple `r`-fields, what we don't support and this PR implements is multiple hops per-`r`-field ?

If so I think you could rename `last_hops` to `r_fields`, `routes` to the singular route as each r field constitutes a revealed forward route, and `last_hop` to `hop` only as it's not necessarily the latest link of the route anymore ?

Reply:
`r_fields` is pretty opaque, maybe `paths_to_payee`?